components/script/parse/xml.rs

```diff
@@ -75,7 +77,13 @@ impl<'a> TreeSink for servoxmlparser::Sink {
 
     fn append(&mut self, parent: JS<Node>, child: NodeOrText<JS<Node>>) {
         let child = match child {
-            NodeOrText::AppendNode(n) => Root::from_ref(&*n),
+            NodeOrText::AppendNode(n) => {
+                let script = n.downcast::<HTMLScriptElement>();
+                if let Some(script) = script {
```

Comment:
To clarify @Ms2ger's answer, it seems to be common convention to have `Option::map` *not* modify any internal state within the closure.